File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOptBinaryValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.OptBinaryInValue.OptBinaryValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOptBinaryValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestOptBinaryValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                byte[].class,
                OptBinaryValue.class.getDeclaredField("optBinary").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                byte[].class,
                OptBinaryValue.class.getMethod("getOptBinary").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                OptBinaryValue.class,
                OptBinaryValue.class.getMethod("setOptBinary", byte[].class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertNull(new OptBinaryValue().getOptBinary());
    }
}

File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOptBinaryValue.scala
package com.dslplatform.ocd.values

import OptBinaryInValue.OptBinaryValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOptBinaryValue extends SpecExtensions {
  def `Test property field type: Option[Array[Byte]]`() =
    checkType(
      typeOf[Option[Array[Byte]]]
    , typeOf[OptBinaryValue].member("optBinary": TermName).asMethod.returnType
    )

  def `Test property default value: None`() {
    assert(OptBinaryValue().optBinary === None)
  }
}
